git/git.go

```diff
@@ -80,6 +96,19 @@ func ResolveRef(ref string) (*Ref, error) {
 	return fullref, nil
 }
 
+func ResolveRefs(refnames []string) ([]*Ref, error) {
+	refs := make([]*Ref, len(refnames))
+	for i, name := range refnames {
+		ref, err := ResolveRef(name)
+		if err != nil {
+			return refs, err
+		}
+
+		refs[i] = ref
+	}
+	return refs, nil
+}
```

Comment:
Yeah this should be a single exec otherwise there's little point of it existing, calling code could loop on ResolveRef. Also (moot if changed) I don't think it should early-return on failure with the separate execs but should tolerate intermediate errors in case caller just wants to deal with the ones that work & skip failures. That would imply a map return value.